src/Event.tsx

```diff
@@ -28,20 +28,37 @@ interface EventProps {
 }
 
 interface EventState {
-  event?: core.Event
+  event?: core.Event,
+  grayedOut: boolean
 }
 
 // TODO: Need to make this component auto-refresh
 class Event extends React.Component<EventProps, EventState> {
   constructor(props: EventProps) {
     super(props)
-    this.state = {}
+    this.state = {
+      grayedOut: true
+    }
   }
 
   async componentDidMount(): Promise<void> {
+    const eventData = await getClient().core().events().get(this.props.id)
+    const projectId = eventData.projectID
+
     this.setState({
-      event: await getClient().core().events().get(this.props.id)
+      event: eventData
     })
+
+    if(projectId) {
+      const authzList = await getClient().core().projects().authz().roleAssignments().list(projectId)
```

Comment:
Let's call `getClient().ocre()` just once at the start of this function and save it in a const named `coreClient`.

Reply:
Also, you can filter results by user, which should give you a lot less to iterate over.

```
const authzList = getClient().core().projects().authz().roleAssignments().list(
  projectId,
  {
    principal: {
      type: authz.PrincipalTypeUser,
      id: await getUserID()
    }
  }
)
```

Reply:
Done ! Will be there in the next commit.